jbpm-flow/src/main/java/org/jbpm/process/instance/context/variable/VariableScopeInstance.java

```diff
@@ -162,5 +168,41 @@ public void setContextInstanceContainer(ContextInstanceContainer contextInstance
     		this.variableInstanceIdPrefix = ((CompositeContextNodeInstance) contextInstanceContainer).getUniqueId();
     	}
 	}
+    
+    public void enforceRequiredVariables() {
+        VariableScope variableScope = getVariableScope();
+        for (Variable variable : variableScope.getVariables()) {
+            String name = variable.getName();
+            if (variableScope.isRequired(name)) {  
```

Comment:
For reducing indentation, it can return if condition is opposite
```suggestion
            if (!variableScope.isRequired(name)) {
              return;
            }  
```

Reply:
it cannot return as it must check all variables

Reply:
you're right, I should have said "continue"

Reply:
if you don't mind I'd prefer to keep just single control flow

Reply:
ok